src/synthesize.py

```diff
@@ -77,6 +77,24 @@ def lazy_synthesize(gdfs: dict[str, DataFrame], target: DataFrame) -> None | Can
     print(out or 'No program found!')
     return out
 
+def lazy_synthesize_gen(gdfs: dict[str, DataFrame], target: DataFrame):
+    ''' Generates the minimum number of programs to get the first that 
+        matches `target` over `gdfs`
+    '''
+    if not isinstance(gdfs, GdfBindings):
+        gdfs = GdfBindings(gdfs)
+
+    checker = make_candidate_filter(gdfs, target)
+    synthesizer = program(gdfs)
+    candidate = next(synthesizer)
```

Comment:
Calling `next` without a default here and on line 96 will throw a `StopIteration` Error when gdfs is empty. 

Proposing changes to streamline iteration.